sdk/src/impl/position-impl.ts

```diff
@@ -183,10 +188,175 @@ export class PositionImpl implements Position {
     return txBuilder;
   }
 
+  async collectFees(
+    resolveATA: boolean = true,
+    updateFeesAndRewards: boolean = true,
+    destinationWallet?: Address,
+    positionWallet?: Address,
+    ataPayer?: Address
+  ): Promise<TransactionBuilder> {
+    const [destinationWalletKey, positionWalletKey, ataPayerKey] = AddressUtil.toPubKeys([
+      destinationWallet ?? this.ctx.wallet.publicKey,
+      positionWallet ?? this.ctx.wallet.publicKey,
+      ataPayer ?? this.ctx.wallet.publicKey,
+    ]);
+
+    const whirlpool = await this.ctx.fetcher.getPool(this.data.whirlpool);
+    if (!whirlpool) {
+      throw new Error(
+        `Unable to fetch whirlpool (${this.data.whirlpool}) for this position (${this.address}).`
+      );
+    }
+
+    let txBuilder = new TransactionBuilder(this.ctx.provider.connection, this.ctx.provider.wallet);
+
+    const accountExemption = await this.ctx.fetcher.getAccountRentExempt();
+    const { ataTokenAddresses, resolveAtaIxs } = await resolveAtaForMints(this.ctx, {
+      mints: [whirlpool.tokenMintA, whirlpool.tokenMintB],
+      accountExemption,
+      receiver: destinationWalletKey,
+      payer: ataPayerKey,
+    });
+
+    const tokenOwnerAccountA = ataTokenAddresses[whirlpool.tokenMintA.toBase58()];
+    const tokenOwnerAccountB = ataTokenAddresses[whirlpool.tokenMintB.toBase58()];
+
+    if (resolveATA) {
+      txBuilder.addInstructions(resolveAtaIxs);
+    }
+
+    const positionTokenAccount = await deriveATA(positionWalletKey, this.data.positionMint);
+
+    if (updateFeesAndRewards) {
+      const updateIx = await this.updateFeesAndRewards();
+      txBuilder.addInstruction(updateIx);
+    }
+
+    const ix = collectFeesIx(this.ctx.program, {
+      whirlpool: this.data.whirlpool,
+      position: this.address,
+      positionTokenAccount,
+      tokenOwnerAccountA,
+      tokenOwnerAccountB,
+      tokenVaultA: whirlpool.tokenVaultA,
+      tokenVaultB: whirlpool.tokenVaultB,
+      positionAuthority: positionWalletKey,
+    });
+
+    txBuilder.addInstruction(ix);
+
+    return txBuilder;
+  }
+
+  async collectRewards(
+    resolveATA: boolean = true,
+    updateFeesAndRewards: boolean = true,
+    destinationWallet?: Address,
+    positionWallet?: Address,
+    ataPayer?: Address,
+    rewardsToCollect?: Address[]
+  ): Promise<TransactionBuilder> {
+    const [destinationWalletKey, positionWalletKey, ataPayerKey] = AddressUtil.toPubKeys([
+      destinationWallet ?? this.ctx.wallet.publicKey,
+      positionWallet ?? this.ctx.wallet.publicKey,
+      ataPayer ?? this.ctx.wallet.publicKey,
+    ]);
+
+    const whirlpool = await this.ctx.fetcher.getPool(this.data.whirlpool);
```

Comment:
While it's quite unlikely whirlpool reward initialize state changes that often, the user might want to always fetch the latest whirlpool stats here. We can add a refresh flag?